sqlf.go

```diff
@@ -70,6 +72,52 @@ func (q *Query) Args() []interface{} {
 	return q.args
 }
 
+// Render returns a string that can be executed directly as a SQL query.
+func (q *Query) Rneder(binder BindVar) (string, error) {
+	s := q.Query(binder)
+	for i, arg := range q.Args() {
+		argString, err := argToString(arg)
+		if err != nil {
+			return "", err
+		}
+
+		// TODO there's a bug where if the arg contains valid binders,
+		// they too will be unintentionally replaced by subsequent arguments.
+
+		// TODO support other BindVars such as `?`.
+
+		// TODO there must be a better way to do this, perhaps using `q.fmt`.
+		s = strings.ReplaceAll(s, fmt.Sprintf("$%d", i+1), argString)
+	}
+	return s, nil
+}
+
+func argToString(arg interface{}) (string, error) {
+	switch arg := arg.(type) {
+	case string:
+		return fmt.Sprintf("'%s'", escapeQuotes(arg)), nil
+	case *pq.StringArray:
```

Comment:
rather than pulling in the `pq` dependency, can you just do type check against `interface{Value()(interface{}, error)}`?

Reply:
Good idea 💡